plugin/main/src/kotlinx/benchmark/gradle/AndroidMultiplatformTasks.kt

```diff
@@ -0,0 +1,219 @@
+            }
+        }
+        doLast {
+            val generatedAndroidProjectDir = generatedAndroidProjectDir(target, compilation)
+            logger.info("Setting up an empty Android project at $generatedAndroidProjectDir")
+
+            generatedAndroidProjectDir.resolve("microbenchmark/build.gradle.kts").let {
+                val unpackedDir = getUnpackAarDir(compilation)
+                val newText = it.readText().replace(
+                    "<<BENCHMARK_CLASSES_JAR_PATH>>",
+                    unpackedDir.resolve("classes.jar").absolutePath.replace("\\", "/")
+                )
+                it.writeText(newText)
+            }
+        }
+    }
+}
+
+private fun Project.createUnpackAarTask(target: AndroidBenchmarkTarget, compilation: KotlinJvmAndroidCompilation) {
+    task<DefaultTask>("unpack${compilation.name.capitalize(Locale.ROOT)}Aar") {
+        group = "benchmark"
+        description = "Unpacks the AAR file produced by ${target.name} compilation '${compilation.name}'"
+        dependsOn("bundle${compilation.name.capitalize(Locale.ROOT)}Aar")
+        doLast {
+            logger.info("Unpacking AAR file produced by ${target.name} compilation '${compilation.name}'")
+
+            val aarFile = getAarFile(compilation)
+
+            if (!aarFile.exists()) {
+                throw IllegalStateException("AAR file not found: ${aarFile.absolutePath}")
+            }
+
+            // TODO: Register the unpacked dir as an output of this task
+            // TODO: Delete the directory if exists before unpacking
+            unpackAarFile(aarFile, compilation)
+        }
+    }
+}
+
+private fun generateSourcesTaskName(target: AndroidBenchmarkTarget, compilation: KotlinJvmAndroidCompilation): String {
+    return "${target.name}${compilation.name.capitalize(Locale.ROOT)}${BenchmarksPlugin.BENCHMARK_GENERATE_SUFFIX}"
+}
+
+private fun Project.createAndroidBenchmarkGenerateSourceTask(target: AndroidBenchmarkTarget, compilation: KotlinJvmAndroidCompilation) {
+    task<DefaultTask>(generateSourcesTaskName(target, compilation)) {
+        group = "benchmark"
+        description = "Generates Android source files for ${target.name} compilation '${compilation.name}'"
+        dependsOn("unpack${compilation.name.capitalize(Locale.ROOT)}Aar")
+        dependsOn("setup${compilation.name.capitalize(Locale.ROOT)}AndroidProject")
+
+        doLast {
+
+            val unpackedDir = getUnpackAarDir(compilation)
+            processClassesJar(unpackedDir, compilation) { classDescriptors ->
+                val targetDir = generatedAndroidProjectDir(target, compilation)
+                    .resolve("microbenchmark/src/androidTest/kotlin")
+
+                targetDir.mkdirs()
+
+                generateBenchmarkSourceFiles(targetDir, classDescriptors)
+            }
+        }
+    }
+}
+
+private fun detectAndroidDevice() {
+    println("Detect running Android devices...")
+    val devices = ProcessBuilder("adb", "devices")
```

Comment:
`adb` may not be on a `PATH`.
In that case it makes sense to check `sdk.dir` property, `ANDROID_HOME` env variable, or , what seems to be a proper way,  [ask Android plugin for an ADB path](https://developer.android.com/reference/tools/gradle-api/8.3/null/com/android/build/api/dsl/SdkComponents#getAdb()).